pkg/services/ngalert/state/historian/encode.go

```diff
@@ -16,9 +16,9 @@ import (
 
 type JsonEncoder struct{}
 
-func (e JsonEncoder) encode(s []stream) ([]byte, error) {
+func (e JsonEncoder) encode(s []Stream) ([]byte, error) {
```

Comment:
WDYT about publicizing the Loki HTTP client and its constructs (stream, queryRes, etc) in a separate PR, first? The churn of making things public is what added a huge volume of change to the current PR, separating this out would make this change more reviewable. Large PRs directly reduce the quality of the review.

I could see this client becoming a totally separate package, even, if we're using it in multiple places. It's written to be a bit independent of historian already.